### ISSUE TITLE
🦠 Model Request: DeepDL

### ISSUE THREAD

**Issue by miquelduranfrigola** [ersilia-os/ersilia#1254]
### Model Name

Drug-likeness scoring based on unsupervised learning

### Model Description

This model evaluates drug-likeness using an unsupervised learning approach, eliminating the need for labeled data and avoiding biases from incomplete negative sets. It extracts features directly from known drug molecules, identifying common characteristics through a recurrent neural network (RNN) language model. By representing molecules as SMILES strings, the model learns the probability distribution of known drugs and assesses new molecules based on their likelihood of appearing in this space. This method offers a robust and consistent evaluation of drug-likeness, outperforming traditional supervised models like TCC in generalization across various datasets.

### Slug

deep-dl

### Tag

Drug-likeness

### Publication

https://pubs.rsc.org/en/content/articlehtml/2022/sc/d1sc05248a

### Source Code

https://github.com/SeonghwanSeo/DeepDL

### License

GPL-3.0-or-later

**Comment by miquelduranfrigola** [comment]
I believe this model can be useful for Day 4 of the AI2050 workshop (Ghana, 2024), based on real-world data received from one of the participants.

**Comment by DhanshreeA** [comment]
@miquelduranfrigola I skimmed through the model code and README, would you have any idea if we want to use the RNN or the GCN weights for incorporating it?

**Comment by miquelduranfrigola** [comment]
Thanks @DhanshreeA - based on the paper, we need to use the RNN weights

**Comment by GemmaTuron** [comment]
@miquelduranfrigola @DhanshreeA just for clarification in case of further questions: 

 _The GCN-based TCC model showed excellent performance when a test set was similar to a training set. However, its performance severely degraded for the other test sets. In contrast, the RNN-based unsupervised learning model showed relatively low data dependency._  From the article discussion

I'll incorporate the RNN

**Comment by GemmaTuron** [comment]
I've incorporated the model @miquelduranfrigola 
Done it in record time of 15 mins - please double check everything is working as it should!

**Comment by GemmaTuron** [comment]
I've updated the model to py3.10 instead of 3.7 as indicated by the authors - this means rdkit and torch are at higher versions, hope this is not messing up with anything. Does not seem so but fyi

**Comment by miquelduranfrigola** [comment]
It works. The only think is that for some molecules it crashes, but Ersilia can deal with it. We can close for now.
